### PR TITLE
fix: Tidy up api_endpoint processing for Text-Embedding Model Tuning demo.

### PR DESCRIPTION
**Pull request by lee1premium**
## Text embedding model tuning

Fixes: b/320696662

## Checklist
- [ X] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [X] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [X] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: generative_ai/embedding_model_tuning.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def tune_embedding_model(
     batch_size: int = 50,
     iterations: int = 300,
 ) -> pipeline_jobs.PipelineJob:
-    match = re.search(r"(.+)(-autopush|-staging)?-aiplatform.+", api_endpoint)
+    match = re.search(r"([\w]+-[\w]+).*-aiplatform.+", api_endpoint)
     location = match.group(1) if match else "us-central1"
```
Comment (dandhlee) [inline generative_ai/embedding_model_tuning.py:38]: nit: consider a bit more global friendly location (unless your API is location restricted) for non-US folks as well

File: generative_ai/embedding_model_tuning.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ def tune_embedding_model(
     batch_size: int = 50,
     iterations: int = 300,
 ) -> pipeline_jobs.PipelineJob:
-    match = re.search(r"(.+)(-autopush|-staging)?-aiplatform.+", api_endpoint)
+    match = re.search(r"([\w]+-[\w]+).*-aiplatform.+", api_endpoint)
```
Comment (dandhlee) [inline generative_ai/embedding_model_tuning.py:37]: Wouldn't this be easier with a string split instead of regex search?

**Review by dandhlee — COMMENTED** [review]
Please use a more descriptive PR title and appropriate type. This seems more like a bug fix than a feature.

File: generative_ai/embedding_model_tuning.py (hunk 2 above)
Comment (lee1premium) [reply generative_ai/embedding_model_tuning.py:37]: I think re.search is cleaner than two splits and a join:

* api_endpoint = "us-central1-xxx-yyy.aiplatform.googleapis.com"
* location1 = re.search(r"^([\w]+-[\w]+)", api_endpoint)[1]
* location2 = "-".join(api_endpoint.split(".")[0].split("-")[:2])  # doesn't look simpler to me.

File: generative_ai/embedding_model_tuning.py (hunk 1 above)
Comment (lee1premium) [reply generative_ai/embedding_model_tuning.py:38]: I guess that 99.9% of the Vertex AI samples would use "us-central1" as the baseline.

Comment (dandhlee) [reply generative_ai/embedding_model_tuning.py:38]: Just a friendly reminder that not everyone needs to stick to one region, and there is a cost/environmental concerns over sticking only to one location. Fine to keep as-is, but if there's no restriction consider using different locations in the future.

File: generative_ai/embedding_model_tuning.py (hunk 2 above)
Comment (lee1premium) [reply generative_ai/embedding_model_tuning.py]: It doesn't work for this case:
* api_endpoint = "us-central1-xxx-yyy.zzzz.aiplatform.googleapis.com"

You have to do two splits and a join to handle other funny cases.

Comment (dandhlee) [reply generative_ai/embedding_model_tuning.py]: Ah. That's icky. Got it.

**Review by dandhlee — APPROVED** [review]
Reviewing for Python samples.

**Comment by lee1premium** [comment]
You should disable FAILED test_gemini_examples.py::test_gemini_safety_config_example

**Pull request closed without merging**